src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternPathSegment.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Routing.Patterns
+{
+    [DebuggerDisplay("{DebuggerToString()}")]
+    public sealed class RoutePatternPathSegment
+    {
+        internal RoutePatternPathSegment(string rawText, RoutePatternPart[] parts)
+        {
+            RawText = rawText;
+            Parts = parts;
+        }
+
+        public bool IsSimple => Parts.Count == 1;
+
+        public IReadOnlyList<RoutePatternPart> Parts { get; }
+
+        public string RawText { get; set; }
```

Comment:
Should this have a public setter?

Reply:
Good catch, this should not be settable.